iceoryx_examples/automotive_soa/include/owl/kom/event_subscriber.inl

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_INL
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_EVENT_SUBSCRIBER_INL
+
+#include "owl/kom/event_subscriber.hpp"
+
+namespace owl
+{
+namespace kom
+{
+template <typename T>
+inline EventSubscriber<T>::EventSubscriber(const core::String& service,
+                                           const core::String& instance,
+                                           const core::String& event) noexcept
+    : m_subscriber({service, instance, event},
+                   {QUEUE_CAPACITY, HISTORY_REQUEST, iox::NodeName_t(), NOT_OFFERED_ON_CREATE})
+{
+}
+
+template <typename T>
+inline void EventSubscriber<T>::Subscribe(std::size_t) noexcept
+{
+    // maxSampleCount is ignored, because it is an argument to the c'tor of m_subscriber as part of SubscriberOptions.
+    // Utilizing late initalization by wrapping m_subscriber in an cxx::optional and calling the c'tor here would be an
+    // option
+    m_subscriber.subscribe();
+}
+
+template <typename T>
+inline void EventSubscriber<T>::Unsubscribe() noexcept
+{
+    m_subscriber.unsubscribe();
```

Comment:
Not allowed to call as soon as `m_subscriber` is attached to the listener, you have to detach it first.

Reply:
See answer [here](https://github.com/eclipse-iceoryx/iceoryx/pull/1355#discussion_r879670570).

Reply:
Please fix this by either adjusting the API or use a smart design pattern to mitigate this. Another possible fix is to attach a `UserTrigger` to the listener which performs the `subscribe` and `unsubscribe`. This is why we introduced this class.

Reply:
Refactored see this [comment](https://github.com/eclipse-iceoryx/iceoryx/pull/1355#discussion_r880340086). Closing this thread and we can continue the discussion in the other one.